qiskit_experiments/data_processing/data_processor.py

```diff
@@ -168,3 +186,75 @@ def train(self, data: List[Dict[str, Any]]):
                         train_data.append(self._call_internal(datum, call_up_to_node=index)[0])
 
                     node.train(train_data)
+
+    def _data_extraction(self, datum: Union[Dict, List[Dict]], call_up_to_node: int = None) -> Any:
```

Comment:
This should format input data to unified format. I think this can be always `List[Dict[str, Any]]` and we can remove branching logic in call to simplify the code.

Reply:
The extraction should handle the output of `ExperimentData.data(...)` which is either `Dict` or `List[Dict]` (see here: https://github.com/Qiskit/qiskit-experiments/blob/be3e21e461ef1a30babea6b4f1e4a718a25abbac/qiskit_experiments/experiment_data.py#L200). The output will always be a list of extracted data. See changes here: 057c38c